### ISSUE TITLE
[Bug] Permute transformation does not permute

### ISSUE THREAD

**Issue by akern40** [pyro-ppl/pyro#2469]
### Issue Description
The `Permute` transformation in `pyro.distributions.transforms` does not perform the expected permutation. From the documentation, the transform "reorders the input dimensions, that is, multiplies the input by a permutation matrix." The code, however, is as follows:
```{python}
 def _call(self, x):
        return x[..., self.permutation]
```
This does not seem to perform a permutation; rather, it selects elements from the last dimension. To reproduce:
```{python}
import torch, pyro
x = torch.rand(2, 3, 4, 5)  # Tensor of shape [2, 3, 4, 5]
p = (3, 2, 1, 0)            # Reverse the ordering of the dimensions
perm = pyro.distributions.transforms.Permute(p)

x.permute(p).shape          # Outputs torch.Size([5, 4, 3, 2])
perm(x).shape               # Outputs torch.Size([2, 3, 4, 4])
```
This is using `torch.__version__ == 1.4.0` and `pyro.__version__ == 1.3.1`.

**Comment by fritzo** [comment]
I think this is a documentation issue and that `transforms.Permute` is intended to "reorder the data along the trailing dimension", in contrast to `torch.permute` which reorders tensor dimensions. @stefanwebb if my interpretation is correct, I believe we merely need to clarify docs, maybe emphasizing the difference from `torch.permute`.

**Comment by martinjankowiak** [comment]
yes that's right. it's about reordering elements within a given dimension. this way if you have normalizing flows that treat, e.g., the first half of the elements differently from the second half, you can get general families of flows by interleaving such "restricted" flows with permutations to get something more general

**Comment by akern40** [comment]
Ah got it - I think, in that case, that my confusion stemmed from a similar situation to #2464. I didn't mean to jump the gun on the pull request, sorry!

**Comment by stefanwebb** [comment]
Thanks for bringing up this point @akern40! I'll add a clarifying note to the docstring for `transforms.Permute` to avoid this confusion in the future

**Comment by stefanwebb** [comment]
I'm going to close this issue for now - @akern40 feel free to reopen if you'd like to continue discussing :)
